DOCSP-45715: Sealed class v3.0 changes (#345)
mongodb/docs-csharp: source/upgrade/v3.txt

    property instead.

- The driver removes individual cluster events from ``MongoClient.Cluster``. To listen for
  cluster events, use `ClusterBuilder.Subscribe() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Core.Configuration.ClusterBuilder.Subscribe.html>`__.

> - The driver seals all concrete serializers. To implement a custom serializer, implement the

Reviewer: We had some questions of "why" classes got sealed.
It's worth adding a general comment followed by these concrete examples:
Something like "Some of the types that were not designed for extension via inheritance became sealed. Starting with 3.0 we recommend using "composition over inheritance" approach for extending the driver functionality."

> - All concrete serializers have been sealed. To implement a custom serializer, implement the
>   ``IBsonSerializer`` interface.
> 
> - The ``MongoClient``, ``MongoDatabase``, and ``MongoCollection`` classes have been
>   sealed. We recommend using the ``IMongoClient``, ``IMongoDatabase``, and

Reviewer 2: S: I'd update to active voice

```suggestion
- The driver seals all concrete serializers. To implement a custom serializer, implement the
  ``IBsonSerializer`` interface.

- The driver seals the ``MongoClient``, ``MongoDatabase``, and ``MongoCollection`` classes. We recommend using the ``IMongoClient``, ``IMongoDatabase``, and
```
